cmd/limactl/shell.go

```diff
@@ -108,7 +108,7 @@ func shellAction(cmd *cobra.Command, args []string) error {
 	}
 	logrus.Debugf("changeDirCmd=%q", changeDirCmd)
 
-	script := fmt.Sprintf("%s ; exec bash --login", changeDirCmd)
+	script := fmt.Sprintf("%s ; exec $SHELL --login", changeDirCmd)
```

Comment:
Does this work with zsh?

Reply:
> Does this work with zsh?

Looks like it:

```console
$ lima
jan@lima-default:/Users/jan$ exec /usr/bin/zsh
lima-default% echo $0
/usr/bin/zsh
```

Reply:
But not for tcsh? (Anybody still uses?)
```console
$ tcsh --login
Unknown option: `--login'
Usage: tcsh [ -bcdefilmnqstvVxX ] [ argument ... ].
```

Reply:
> But not with tcsh? (Anybody still uses?)

The only way to run a command via a `tcsh` login shell seems to be:

```console
echo $CMD | tcsh -l
```

Users still have to install and configure the shell inside the VM (and then reboot) before they can use it:

```console
lima sudo apt install tcsh
lima sudo chsh -s /usr/bin/tcsh $USER
limactl stop
limactl start
```

I guess the problem is that they are now trapped because they can't undo this change anymore. I wonder if we should provide a `limactl shell --bash` option to force the shell to use `bash` instead of the configured default shell?

Reply:
I have read the documentation for [fish](https://fishshell.com/docs/current/cmds/fish.html?highlight=login), [zsh](https://zsh.sourceforge.io/Guide/zshguide02.html) and [bash](https://www.gnu.org/software/bash/manual/html_node/Invoking-Bash.html), all of which explicitly support the '--login' parameter and as an [login shell](https://unix.stackexchange.com/questions/38175/difference-between-login-shell-and-non-login-shell).
It seems that tcsh does not support use as a login shell, so it may not be appropriate to use tcsh as the default shell in this case. But I think you could submit an issue to this project for discussion.

Reply:
> But I think you could submit an issue to this project for discussion.

I don't really care about supporting `tcsh` as a login shell for lima; but I'm concerned that somebody would try it, and gets locked out of their VM. That's why I was suggesting the `--bash` option to be able to still get a shell to fix the issue.

Reply:
But we're just opening a new interactive shell after each ssh session, which shouldn't be a login shell, right?

Reply:
> which shouldn't be a login shell

I'm not sure what you are saying here. It won't read `~/.profile` (or equivalent) unless it is a login shell.

Reply:
<img width="786" alt="image" src="https://user-images.githubusercontent.com/3659110/161499840-ff5fdf6d-69df-4f78-8822-b756c4924fb4.png">
It seems that '-l' is a more generic argument than '--login'.

Reply:
<img width="560" alt="image" src="https://user-images.githubusercontent.com/3659110/161505460-e9b1db11-1b6f-452f-ab9a-7435c55ac4f6.png">

Alternatively, we can consider using the login command to open a new session, which will be opened using the default shell.